js/src/modals/SMSVerification/store.js

```diff
@@ -0,0 +1,236 @@
+// Copyright 2015, 2016 Ethcore (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import { observable, computed, autorun, action } from 'mobx';
+import phone from 'phoneformat.js';
+import { sha3 } from '../../api/util/sha3';
+
+import ABI from '../../contracts/abi/sms-verification.json';
+// TODO: move this to a better place
+const contract = '0xcE381B876A85A72303f7cA7b3a012f58F4CEEEeB';
```

Comment:
Indeed. Do a name lookup against the registry. As it stands, this will fail against Homestead - and as soon as we have merged, it is available for both networks. (Unless we add a switch for only Morden).